### ISSUE TITLE
Revise axis labeling approach

### ISSUE THREAD

**Issue by matthewfeickert** [matthewfeickert/heputils#68]
In [`mplhep` `v0.3.8` ](https://pypi.org/project/mplhep/0.3.8/) there was a change in how `mplhep` operates with labels.

In the `dev-example` notebook, the following now updates the `x`-axis label to the default `"xaxis"`

```python
fig, ax = plt.subplots()
ax = heputils.plot.stack_hist(
    simulation_hists,
    labels=labels,
    color=colormap,
    xlabel=r"$X$ Mass [GeV]",
    ylabel="Count",
    scale_factors=scale_factors,
    ax=ax,
)
ax = heputils.plot.data_hist(data_hist, ax=ax)
```

![example_fail](https://user-images.githubusercontent.com/5142394/127728564-9dd342ea-2a69-46c5-8f70-b50942905ef7.png)

but passing a `data_hist` still works

```python
fig, ax = plt.subplots()
ax = heputils.plot.stack_hist(
    simulation_hists,
    labels=labels,
    color=colormap,
    xlabel=r"$X$ Mass [GeV]",
    ylabel="Count",
    scale_factors=scale_factors,
    data_hist=data_hist,
    ax=ax,
)
```

![example_pass](https://user-images.githubusercontent.com/5142394/127728593-4dbe80fe-2d20-48e5-9544-64528aac2146.png)

So maybe it is necessary to check if

```python
ax.get_xlabel() != "xaxis"
```

during axis updates.

This means that

https://github.com/matthewfeickert/heputils/blob/2d077ea9597f5e2fdffb1a93bab78a9c3f57beba/setup.cfg#L42-L43

should also be updated to

```
mplhep>=0.3.8
hist[plot]>=2.3.0
```

**Comment by matthewfeickert** [comment]
cc @msneubauer as this will be relevant.

**Comment by matthewfeickert** [comment]
This might take some work to fix given that for `mplhep` `v0.3.9`

```python
import hist
import matplotlib.pyplot as plt
import mplhep
import numpy as np
from hist import Hist

if __name__ == "__main__":
    np.random.seed(0)

    hist_1 = Hist(hist.axis.Regular(10, 0.0, 1.0, name="x"))
    hist_1.fill(np.random.random(1000))

    fig, ax = plt.subplots()
    mplhep.histplot(hist_1, ax=ax)
    ax.set_xlabel("my label")  # want this to show up

    hist_2 = Hist(hist.axis.Regular(10, 0.0, 1.0, name="x"))
    hist_2.fill(np.random.random(1000))

    # This overwrite the pervious label with the value of hist_2.axes[0].label, which is "x"
    mplhep.histplot(hist_2, ax=ax)  # apparently the ax label info isn't used here?
    fig.savefig("example.png", facecolor="white")
```

gives

![example](https://user-images.githubusercontent.com/5142394/128290422-b5386dba-0d75-4f3a-b6c8-b4b653ff97a5.png)
